kroxylicious-kubernetes-api/src/main/resources/META-INF/fabric8/kafkaproxyingresses.kroxylicious.io-v1.yml

```diff
@@ -69,11 +69,20 @@ spec:
                     clusterIP specifies that this ingress is for access within the same Kubernetes cluster using 
                     ClusterIP Kubernetes Services.
                   required: [ "protocol" ]
+                  x-kubernetes-validations:
+                    - message: "nodeIdentifiedBy mechanism 'serverNameIndication' is incompatible with TCP protocol"
+                      rule: "!has(self.nodeIdentifiedBy) || !(self.nodeIdentifiedBy == 'serverNameIndication' && self.protocol == 'TCP')"
                   properties:
                     protocol:
                       description: "Protocol specifies the network protocol this ingress expects to receive."
                       type: string
                       enum: ["TCP", "TLS"]
+                    nodeIdentifiedBy:
+                      type: string
+                      description: "NodeIdentifiedBy specifies the mechanism used to identify upstream Kafka nodes. If serverNameIndication
+                      is selected, the ingress protocol must be TLS"
+                      default: "port"
+                      enum: ["port", "serverNameIndication"]
```

Comment:
in the proxy config we use `sniHostIdentifiesNode: `, maybe the option should be called `sniHost`

Reply:
I think the proxy should probably use `sniIdentifiesNode` though the `iI` transition is hard on the eye too but to answer your actual question if we don't like `sni` (and I can see arguments both ways there) is `sniHost` any different? 

`enum: ["port", "sni"]` would work for me. 

```yaml
kind: KafkaProxyIngress
apiVersion: kroxylicious.io/v1alpha1
metadata:
  name: cluster-ip
  namespace: my-proxy
spec:
  proxyRef:
    name: simple
  clusterIP:
    protocol: TLS
    nodeIdentifiedBy: sni
 ```
and
```yaml
kind: KafkaProxyIngress
apiVersion: kroxylicious.io/v1alpha1
metadata:
  name: cluster-ip
  namespace: my-proxy
spec:
  proxyRef:
    name: simple
  clusterIP:
    protocol: TLS
    nodeIdentifiedBy: port
 ```

Reply:
I'm happy enough with `sni`.

Maybe having it capitalized like `SNI`, would look nicer sitting next to the `TLS` protocol?

Reply:
+1 to capitalising SNI.

Reply:
pushed up SNI